core/meterer/meterer.go

```diff
@@ -356,6 +344,43 @@ func ValidateQuorum(headerQuorums []uint8, allowedQuorums []uint8) error {
 	return nil
 }
 
+// ValidateReservations ensures that the quorums listed in the blobHeader are present within allowedQuorums
```

Comment:
Maybe something like this instead is more clear
```suggestion
// ValidateReservations ensures that the quorums listed in the blobHeader are present within allowedQuorums.
// 
// Parameters:
//   - timestamp: time in nanoseconds
// 
// Notes:
//   - Reservations that don't use all allowed quorums are still accepted
//   - Charges apply to ALL allowed quorums, even if not all are used
//   - On-demand requests have special requirements: they must use ETH and EIGEN quorums only
```